lib/compile.js

```diff
@@ -36,29 +36,55 @@ function compileLoad(loader, load, opts, compilers) {
   });
 }
 
+function retrieveCachedOutput(cache, load) {
+  if (! cache)
+    return;
+
+  var entry = cache[load.name];
+  if (! entry)
+    return;
+
+  return load.source === entry.source && entry.output;
```

Comment:
`entry.output` isn't set anywhere?

Perhaps we should do a hash of the input source for the cache object instead of storing the full source in the cache.